### PR TITLE
feat(issues): filter by category

### LINKED ISSUE #19: Add issues filtering by category

### PR DESCRIPTION
**Pull request by mapra99**
Closes #19 

Mobile:

https://github.com/mapra99/feedbackr/assets/27536621/8e054620-c071-4d40-9043-575b652b6b69

Desktop:

https://github.com/mapra99/feedbackr/assets/27536621/4776358a-c7f3-40b5-9c5e-d1f9b7923347

### CODE REVIEW

File: api/app/services/issues_feed_builder.rb [hunk 1]
```diff
@@ -32,4 +34,10 @@ def sorted
       raise StandardError, 'Invalid sort params'
     end
   end
+
+  def filtered
+    return issues if filter_params.blank? || filter_params[:category].blank?
+
+    issues.filter_by_categories(filter_params[:category].split(','))
+  end
```
Comment (ImMPrada) [inline api/app/services/issues_feed_builder.rb:42]: <3

File: api/spec/services/issues_feed_builder_spec.rb [hunk 2]
```diff
@@ -36,5 +38,19 @@
...
+
+      before do
+        create(:issue, issue_category: category, product:)
+        create(:issue, product:)
+      end
+
+      it 'returns issues filtered by categories' do
+        expect(result.pluck(:issue_category_id)).to eq [category.id]
+      end
+    end
```
Comment (ImMPrada) [inline api/spec/services/issues_feed_builder_spec.rb:54]: does 'context' the same of 'describe' ?

what about border case: `return issues if filter_params.blank? || filter_params[:category].blank?`

File: web/src/components/categories-filter/categories-filter.tsx [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  const [selection, setSelection] = useState<Set<string>>(new Set(filterParams.category ? filterParams.category.split(',') : ["all"]))
+  const router = useRouter()
+
+  const handleSelectionChange = (name: string) => {
+    const updatedSelection = new Set(updateSelection(name))
+    updatedSelection.delete("all")
+
+    const newPath = updateCurrentPath({ category: Array.from(updatedSelection).join(",") })
+    router.push(newPath)
+  }
```
Comment (ImMPrada) [inline web/src/components/categories-filter/categories-filter.tsx:22]: so 

- if I click a category:
  - 'All' will be unslected
  - clicking another category this last will be added to the filters
  - selecting all categories will transform the selection to 'All'

- If I have, for example 2 of 5 filteres, selected, and click 'All', the selected filters will be deleted, and keep 'All' selected

- if I don't have any filter selected, and I click 'All', 'All' won't be deleted


right?

**Review by ImMPrada — APPROVED** [review]
questions

File: api/spec/services/issues_feed_builder_spec.rb (hunk 2 above)
Comment (mapra99) [reply api/spec/services/issues_feed_builder_spec.rb:54]: > does 'context' the same of 'describe' ?

yeah, it is the same thing

about the edge case, I think it's being covered by the other context that tests when sorting is passed because in that case filters are not being passed hahah

File: web/src/components/categories-filter/categories-filter.tsx (hunk 3 above)
Comment (mapra99) [reply web/src/components/categories-filter/categories-filter.tsx:22]: Correct, except for this one:
- selecting all categories will transform the selection to 'All'

Selecting all categories will just keep all categories selected. 'All' won't be selected automatically
